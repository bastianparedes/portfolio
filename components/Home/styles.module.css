.home {
  position: relative;
  height: 100vh;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background: linear-gradient(rgb(0, 0, 94), rgb(0, 0, 0));
}

.cuadro {
  color: rgb(255, 255, 255);
  text-align: center;
  backdrop-filter: blur(0.625rem);
  user-select: none;
}

.myName {
  font-size: 6rem;
  margin: 1rem auto;
}

.typewriterContainer {
  width: fit-content;
  margin: 0 auto 1rem auto;
  padding: 0 0.1875rem;
  color: rgb(255, 255, 255, 1);
  font-size: 3rem;
  font-weight: bold;
}

.btnDown {
  position: relative;
  width: 5rem;
  height: 5rem;
  margin-top: 1rem;
  background-color: rgb(255, 255, 255);
  color: rgb(0, 0, 0);
  border-radius: 50%;
  transition: all 0.3s ease;
  text-decoration: none;
  font-size: 3.5rem;
  outline: inherit;
  display: flex;
  align-items: center;
  justify-content: center;
}

.btnDown:hover {
  transform: scale(1.1);
  background-color: #5196fd;
  color: rgb(255, 255, 255);
}

@media screen and (max-width: 768px) {
  .myName {
    font-size: 5rem;
  }
  .typewriterContainer {
    font-size: 2rem;
  }
}

@media screen and (max-width: 480px) {
  .myName {
    font-size: 4rem;
  }
  .typewriterContainer {
    font-size: 2.5rem;
  }
  .btnDown {
    width: 3rem;
    height: 3rem;
  }
}
