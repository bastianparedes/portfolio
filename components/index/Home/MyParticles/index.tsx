import React, { useCallback } from 'react';
import Particles from 'react-tsparticles';
import { loadFull } from 'tsparticles';
import type { Engine } from 'tsparticles-engine';
import styles from './styles.module.scss';

const MyParticles = (): JSX.Element => {
  const particlesInit = useCallback(async (engine: Engine) => {
    await loadFull(engine);
  }, []);

  return (
    <Particles
      className={styles.particles}
      init={particlesInit}
      options={{
        autoPlay: true,
        fullScreen: false,
        fpsLimit: 60,
        interactivity: {
          detectsOn: 'canvas',
          events: {
            onHover: {
              enable: true,
              mode: 'grab'
            },
            onClick: {
              enable: true,
              mode: 'push'
            }
          },
          modes: {
            push: {
              quantity: 1
            },
            grab: {
              distance: 200
            }
          }
        },
        background: {},
        particles: {
          color: {
            value: '#2034fa'
          },
          number: {
            value: 30,
            limit: 30
          },
          move: {
            enable: true,
            direction: 'top',
            random: true,
            speed: 2,
            trail: {
              enable: false
            },
            vibrate: false
          },
          size: {
            value: {
              min: 1,
              max: 5
            },
            animation: {
              enable: true,
              speed: 5
            }
          },
          shape: {
            type: 'circle'
          },
          links: {
            color: '#2034fa',
            distance: 200,
            enable: true,
            width: 2
          }
        },
        pauseOnBlur: true,
        pauseOnOutsideViewport: true
      }}
    />
  );
};

export default MyParticles;
