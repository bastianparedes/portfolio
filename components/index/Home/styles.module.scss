@import 'bastianparedes/styles/_breakpoints';

.home {
  position: relative;
  height: 100vh;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  background: linear-gradient(rgb(0, 0, 94), rgb(0, 0, 0));
}

.cuadro {
  color: rgb(255, 255, 255);
  text-align: center;
  backdrop-filter: blur(0.625rem);
  user-select: none;
}

.myName {
  font-size: 6rem;
  margin: 1rem auto;
}

.typewriterContainer {
  width: fit-content;
  margin: 0 auto 1rem auto;
  padding: 0 0.1875rem;
  color: rgb(255, 255, 255, 1);
  font-size: 3rem;
  font-weight: bold;
}

.btnDown {
  position: relative;
  width: 5rem;
  height: 5rem;
  margin-top: 1rem;
  background-color: rgb(255, 255, 255);
  color: rgb(0, 0, 0);
  border-radius: 50%;
  transition: all 0.3s ease;
  text-decoration: none;
  font-size: 3.5rem;
  outline: inherit;
  display: flex;
  align-items: center;
  justify-content: center;

  &:hover {
    transform: scale(1.1);
    background-color: #5196fd;
    color: rgb(255, 255, 255);
  }
}

@include phone-and-tablet {
  .myName {
    font-size: 5rem;
  }
  .typewriterContainer {
    font-size: 2rem;
  }
}

@include phone {
  .myName {
    font-size: 4rem;
  }
  .typewriterContainer {
    font-size: 2.5rem;
  }
  .btnDown {
    width: 3rem;
    height: 3rem;
  }
}
