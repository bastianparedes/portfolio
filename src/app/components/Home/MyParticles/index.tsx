import React, { useCallback } from 'react';

import Particles from 'react-tsparticles';
import { loadFull } from 'tsparticles';
import type { Engine } from 'tsparticles-engine';

import styles from './styles.module.scss';

const MyParticles = () => {
  const particlesInit = useCallback(async (engine: Engine) => {
    await loadFull(engine as any);
  }, []);

  return (
    <Particles
      className={styles.particles}
      init={particlesInit}
      options={{
        autoPlay: true,
        background: {},
        fpsLimit: 60,
        fullScreen: false,
        interactivity: {
          detectsOn: 'canvas',
          events: {
            onClick: {
              enable: true,
              mode: 'push'
            },
            onHover: {
              enable: true,
              mode: 'grab'
            }
          },
          modes: {
            grab: {
              distance: 200
            },
            push: {
              quantity: 1
            }
          }
        },
        particles: {
          color: {
            value: '#2034fa'
          },
          links: {
            color: '#2034fa',
            distance: 200,
            enable: true,
            width: 2
          },
          move: {
            direction: 'top',
            enable: true,
            random: true,
            speed: 2,
            trail: {
              enable: false
            },
            vibrate: false
          },
          number: {
            limit: 30,
            value: 30
          },
          shape: {
            type: 'circle'
          },
          size: {
            animation: {
              enable: true,
              speed: 5
            },
            value: {
              max: 5,
              min: 1
            }
          }
        },
        pauseOnBlur: true,
        pauseOnOutsideViewport: true
      }}
    />
  );
};

export default MyParticles;
