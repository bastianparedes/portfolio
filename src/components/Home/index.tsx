import React from 'react'
import Particles from 'react-tsparticles'

import styles from './styles.module.css'
import { BiDownArrowAlt } from 'react-icons/bi'
import Typewriter from 'typewriter-effect'

const Home = (): JSX.Element => {
  return (
    <section id="home" className={styles.home}>
      <Particles
        className={styles.particles}
        options={{
          autoPlay: true,
          fullScreen: false,
          background: {},
          fpsLimit: 60,
          interactivity: {
            detectsOn: 'canvas',
            events: {
              onHover: {
                enable: true,
                mode: 'grab'
              },
              onClick: {
                enable: true,
                mode: 'push'
              }
            },
            modes: {
              push: {
                quantity: 1
              },
              grab: {
                distance: 200
              }
            }
          },
          particles: {
            color: {
              value: 'rgb(35, 52, 250)'
            },
            number: {
              value: 30,
              limit: 30
            },
            move: {
              enable: true,
              direction: 'top',
              random: true,
              speed: 2,
              trail: {
                enable: false
              },
              vibrate: false
            },
            size: {
              value: {
                min: 1,
                max: 5
              },
              animation: {
                enable: true,
                speed: 5
              }
            },
            shape: {
              type: 'circle'
            },
            links: {
              color: 'rgb(35, 52, 250)',
              distance: 200,
              enable: true,
              width: 2
            }
          },
          pauseOnBlur: true,
          pauseOnOutsideViewport: true
        }}
      />
      <div className={styles.cuadro}>
        <h1 className={styles.myName}>Basti√°n Paredes</h1>
        <div className={styles.typewriterContainer}>
          <Typewriter
            onInit={(typewriter) => {
              typewriter.typeString('Full Stack Developer').start()
            }}
          />
        </div>
      </div>
      <a className={styles.btnDown} href="#about">
        <BiDownArrowAlt />
      </a>
    </section>
  )
}

export default Home
