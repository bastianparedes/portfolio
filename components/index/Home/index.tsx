import React from 'react'
import styles from './styles.module.scss'
import { BiDownArrowAlt } from 'react-icons/bi'
import Typewriter from 'typewriter-effect'
import constants from '../../../config/constants.json'
import MyParticles from './MyParticles'

const Home = (): JSX.Element => {
  return (
    <section id={constants.NAV.BUTTONS.HOME.ID} className={styles.home}>
      <MyParticles />
      <div className={styles.cuadro}>
        <h1 className={styles.myName}>{constants.HOME.NAME}</h1>
        <div className={styles.typewriterContainer}>
          <Typewriter
            onInit={(typewriter) => {
              typewriter.typeString(constants.HOME.JOB).start()
            }}
          />
        </div>
      </div>
      <a className={styles.btnDown} href={'#' + constants.NAV.BUTTONS.ABOUT.ID}>
        <BiDownArrowAlt />
      </a>
    </section>
  )
}

export default Home
