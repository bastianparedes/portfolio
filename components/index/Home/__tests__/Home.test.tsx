import React from 'react';
import Home from '..';
import { render } from '@testing-library/react';

interface TypewriterProps {
  onInit: (typewriter: {
    typeString: () => {
      start: () => null;
    };
  }) => void;
}

jest.mock('typewriter-effect', () => ({ onInit }: TypewriterProps) => {
  const typewriter = {
    typeString: () => ({
      start: () => null
    })
  };

  onInit(typewriter);
});
jest.mock('../MyParticles', () => () => <></>);

describe('<Home />', () => {
  it('should render', () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
