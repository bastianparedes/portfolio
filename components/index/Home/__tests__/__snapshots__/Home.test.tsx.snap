// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render 1`] = `
<div>
  <section
    class="home"
    id="home"
  >
    <div
      class="cuadro"
    >
      <h1
        class="myName"
      >
        Basti√°n Paredes
      </h1>
      <div
        class="typewriterContainer"
      />
    </div>
    <a
      class="btnDown"
      href="#about"
    />
  </section>
</div>
`;
